runtime/jailer_integ_test.go

```diff
@@ -78,24 +81,31 @@ func testJailer(t *testing.T, jailerConfig *proto.JailerConfig) {
 	stdout := startAndWaitTask(ctx, t, c)
 	require.Equal("hello", stdout)
 
-	defer func() {
-		err := c.Delete(ctx, containerd.WithSnapshotCleanup)
-		require.NoError(err, "failed to delete a container")
-	}()
+	stat, err := os.Stat(filepath.Join(shimBaseDir(), "default", vmID))
+	require.NoError(err)
+	assert.True(t, stat.IsDir())
+
+	err = c.Delete(ctx, containerd.WithSnapshotCleanup)
+	require.NoError(err, "failed to delete a container")
+
+	_, err = fcClient.StopVM(ctx, &proto.StopVMRequest{VMID: vmID})
+	require.NoError(err)
+
+	_, err = os.Stat(filepath.Join(shimBaseDir(), "default", vmID))
```

Comment:
So this will leave the ShimBaseDir()/default dir?

What happens if we are using multiple namespaces? Do those shimBaseDir/Namespace dirs ever get cleaned up?

Reply:
Yes. The namespace directory will be kept. One thing we could do for workaround the issue is creating the directory like `${namespace}-${vmid}`, not '${namespace}/${vmid}', but not so sure the value it will add, assuming all directories are empty anyway.

Interestingly containerd seems having the same problem.

```
% sudo ls /var/lib/containerd/io.containerd.runtime.v2.task
default
% sudo ctr -n ns1 run --tty --rm docker.io/library/debian:latest debian
root@ip-172-31-25-63:/# exit
% sudo ls /var/lib/containerd/io.containerd.runtime.v2.task            
default  ns1
% sudo ls /var/lib/containerd/io.containerd.runtime.v2.task/ns1 
%
```

The empty ns1 directory is kept, while there are no containers inside.